jbmc/src/java_bytecode/java_object_factory_parameters.h

```diff
@@ -18,9 +18,20 @@ struct java_object_factory_parameterst final : public object_factory_parameterst
   }
 
   explicit java_object_factory_parameterst(const optionst &options)
-    : object_factory_parameterst(options)
+    : object_factory_parameterst(options),
+      assume_inputs_non_null(false),
+      assume_inputs_interval()
   {
   }
+
+  /// Assume reference (i.e., non-primitive) inputs to be non-null
+  bool assume_inputs_non_null;
```

Comment:
:no_entry_sign: I don't think this belongs here, because
 *  It isn't used in ``java_object_factory.cpp`.
 *  Its purpose overlaps with `min_null_tree_depth`, which `java_object_factory_parameterst` inherits from `object_factory_parameterst`.
 *  Because `java_object_factory_parameterst` is a struct and the two fields are publicly accessible, this data structure cannot enforce any invariants on the combination of `assume_inputs_non_null` and `min_null_tree_depth` making logical sense.

For the above reasons, I am in favour of keeping it as a field of `java_bytecode_languaget`, rather than adding it to `java_object_factory_parameterst`.

Reply:
I'd consider it useful to deprecate `java-assume-inputs-non-null` and replace it by a language-independent `min_null_tree_depth` option.

Reply:
As this is out of the scope of what I originally wanted to do, I will leave the option where it was.